source/extensions/transport_sockets/http_11_proxy/connect.cc

```diff
@@ -29,8 +29,21 @@ bool UpstreamHttp11ConnectSocket::isValidConnectResponse(absl::string_view respo
 
 UpstreamHttp11ConnectSocket::UpstreamHttp11ConnectSocket(
     Network::TransportSocketPtr&& transport_socket,
-    Network::TransportSocketOptionsConstSharedPtr options)
-    : PassthroughSocket(std::move(transport_socket)), options_(options) {
+    Network::TransportSocketOptionsConstSharedPtr options,
+    std::shared_ptr<const Upstream::HostDescription> host, bool legacy_behavior)
+    : PassthroughSocket(std::move(transport_socket)), options_(options),
+      legacy_behavior_(legacy_behavior) {
+  if (legacy_behavior_) {
+    legacyConstructor();
+    return;
+  }
+
+  header_buffer_.add(
+      absl::StrCat("CONNECT ", host->address()->asStringView(), " HTTP/1.1\r\n\r\n"));
+  need_to_strip_connect_response_ = true;
+}
+
+void UpstreamHttp11ConnectSocket::legacyConstructor() {
   if (options_ && options_->http11ProxyInfo() && transport_socket_->ssl()) {
```

Comment:
Apparently, the `proxy_address` is used as a flag to tell this extension to accept non-SSL inner transport socket. 

I think to add a separated flag/configuration would be better.